packages/neos-ui/src/Containers/LeftSideBar/NodeTreeSearchBar/NodeTreeFilter/index.js

```diff
@@ -30,36 +32,49 @@ export default class NodeTreeFilter extends PureComponent {
     }
 
     render() {
-        const {i18nRegistry, nodeTypesRegistry, onChange, value} = this.props;
+        const {i18nRegistry, nodeTypesRegistry, neos, onChange, value} = this.props;
         const label = i18nRegistry.translate('filter', 'Filter', {}, 'Neos.Neos', 'Main');
 
-        const documentNodeTypes = nodeTypesRegistry
-            .getSubTypesOf('Neos.Neos:Document')
-            .map(nodeTypeName => nodeTypesRegistry.getNodeType(nodeTypeName))
-            .filter(i => i);
-        const options = documentNodeTypes.map(nodeType => ({
-            value: nodeType.name,
-            label: i18nRegistry.translate(nodeType.label),
-            icon: nodeType.ui.icon
-        }));
+        const presets = $get('configuration.nodeTree.presets', neos);
+        let options = Object.keys(presets)
+            .filter(presetName => (presetName !== 'default'))
+            .map(presetName => ({
+                    value: presets[presetName].baseNodeType,
+                    label: presets[presetName].ui && presets[presetName].ui.label || '[' + presetName + ']',
+                    icon: presets[presetName].ui && presets[presetName].ui.icon || null,
+                })
+            );
+
+        if (options.length === 0) {
+            const documentNodeTypes = nodeTypesRegistry
+                .getSubTypesOf('Neos.Neos:Document')
```

Comment:
Is it necessary to hardcode `'Neos.Neos:Document'` here? Can't one of the roles be used instead?

Reply:
I mean this: https://github.com/neos/neos-ui/blob/master/Configuration/Settings.yaml#L104

Reply:
This is the old implementation

Reply:
Fair point. Would be a good opportunity to refactor though :)